pkg/ansible/controller/reconcile.go

```diff
@@ -79,6 +72,20 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 		"namespace", u.GetNamespace(),
 	)
 
+	err := r.Client.Get(context.TODO(), request.NamespacedName, u)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			logger.Info("Resource not found, assuming it was deleted", "namespace", request.NamespacedName, "resource", u)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		logger.Error(err, "Failed to get Resource.")
```

Comment:
I don't think it is necessary to log this error, since controller-runtime will also log it when we return from `Reconcile()`.

Ditto for the logged error in line 209.